Dev class assignments

varunasingh opened the pull request:
Method and Model signatures for Views, Entities, Daos, Presenters and Tests.

varunasingh committed:
Clazz Assignments signatures

varunasingh committed:
Added entities to umappdatabase

varunasingh committed:
Clazz Assignment Test signatures

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/controller/ClazzAssignmentDetailPresenter.kt:
```diff
@@ -0,0 +1,65 @@
+        if(arguments.containsKey(ARG_CLAZZ_ASSIGNMENT_UID)){
+            val clazzAssignmentUid = arguments[ARG_CLAZZ_ASSIGNMENT_UID]!!.toLong()
+            GlobalScope.launch {
+                clazzAssignment = clazzAssignmentDao.findWithMetricByUidAsync(clazzAssignmentUid)
```
Better to use ?: 0. We should avoid !!

varunasingh committed:
Clazz assignment changes in Detail and other fixes based on feedback
from pull request

varunasingh committed:
Added note for checking permissions

varunasingh committed:
Commiting progress before merge

varunasingh committed:
Merge branch 'dev-kmp-irc-sprint6' into dev-class-assignments

varunasingh committed:
Added recycler item xmls

varunasingh committed:
Added fragments and strings for clazz assignments

varunasingh committed:
Added edit screen and list

mikedawson committed:
Add ContentEntry picker mode to ContentEntryList.

varunasingh committed:
WIP. All bits in. Need fixed.

varunasingh committed:
WIP. commit before merge

varunasingh committed:
Merge branch 'dev-contententrypicker' into dev-class-assignments

varunasingh committed:
WIP content picker + recyclerview

varunasingh committed:
Added title field for assignments

varunasingh committed:
Enabling sync and all

varunasingh committed:
heading fix

varunasingh committed:
fix start and due date on editing assignment

varunasingh committed:
RV WIP

varunasingh committed:
Fixed npe in userprofile
Added content rv
updated queries for clazz assignment <-> content join lookup

mikedawson committed:
Merge branches 'dev-class-assignments' and 'master' of github.com:UstadMobile/UstadMobile into dev-class-assignments

mikedawson committed:
Fix issue #93

Update Android build tools with Android 3.6 stable release

mikedawson committed:
Minor refactor of ClazzAssignmentDetailAssignmentPresenter

varunasingh committed:
Added features in clazz edit.
Fixed toolbar themese

varunasingh committed:
Merge branch 'dev-class-assignments' of github.com:UstadMobile/UstadMobile into dev-class-assignments

mikedawson committed:
Add ViewNameListFragmentPagerAdapter to simplify Android activities that have tabs (each with their own Fragment / Presenter etc)

mikedawson committed:
Update ViewNameListFragmentPagerAdapterRoboElectricTest to use RoblectricTestRunner and check fragment arguments are set.

varunasingh committed:
Fix with 3.6 android studio

mikedawson committed:
Use ViewNameListFragmentPagerAdapter to run HomeContentEntryTabsFragment

varunasingh committed:
Merge branch 'master' into dev-class-assignments
Added permission for assignments
Changed class and assignment to use common vie pager
Added date picker binder

mikedawson committed:
Merge branch 'dev-class-assignments' of github.com:UstadMobile/UstadMobile into dev-class-assignments

mikedawson committed:
New list model based on databinding on ClazzList2

mikedawson committed:
Fix HttpClient issue for Android 5
Update to build tools 3.6.1

mikedawson committed:
Merge branch 'master' into dev-class-assignments-edit2

varunasingh committed:
Added Date picker binder
jvm Tests begin

varunasingh committed:
Added tests logic

varunasingh committed:
Date and Time binding adapter fix

varunasingh committed:
Addd picture binding adapters for person and contententry

mikedawson committed:
Work in progress change to enable Xapi arguments to be passed to XapiPackage

mikedawson committed:
Work-in-progress - set contentEntryUid on statements being received via embedded XAPI server

mikedawson committed:
Implement updates for XapiPackageContentPresenter to use updated Xapi endpoint

mikedawson committed:
Fix naming on UmAccount. Fix constructor parameters in XapiPackageContentActivity

mikedawson committed:
Improve null-safety in StatementEndpoint. Fix handling statements that have a result but no duration

mikedawson committed:
Handle parsing of the progress extension on Xapi statements

mikedawson committed:
Refactor contentEntryUid on statements to be associated with XObject instead of statement

varunasingh committed:
Merge branch 'master' into dev-class-assignments

mikedawson committed:
Add conentEntryUid to StatementEntity and update tests

varunasingh committed:
Added progress xapi query

varunasingh committed:
Merge branch 'master' into dev-class-assignments

varunasingh committed:
Added progress to assignment detail

mikedawson committed:
Refactor DatePicker ViewBinding adapter

mikedawson committed:
Merge remote-tracking branch 'origin/dev-class-assignments' into dev-class-assignments

mikedawson committed:
Fix dateLong DataBinder (required Tag id Int to be a resource id value)

varunasingh committed:
Picture using RV
updated binding adapters
updated settings icon colors
updated a bug in people list
updated queries for xapi statmeents check for assignments

varunasingh committed:
Merge branch 'dev-class-assignments' of github.com:UstadMobile/UstadMobile into dev-class-assignments

mikedawson commented:
This is obsolete and replaced with #159
